bip-rpob-tx-template.mediawiki

```diff
@@ -0,0 +1,284 @@
+
+===Output 3, Optional: Change===
+<pre>
+value:              (any)
+public_key_script:  1 <32-byte-taproot_witness_program>
+</pre>
+
+*This output MAY have any value.
+*The public key MUST be a standard P2TW output.
+
+===Signature: Included in Output 1===
+<pre>
+rpob_schnorr_signature:       64-bytes, bip-340, signature (
+                                        <32-byte-rpob_signature_message_hash>,
+                                        <32-byte-secp256k1_private_key>
+                                        )
+
+rpob_signature_message_hash:  32-bytes, bip-340, sha256_taged_hash (
+                                        <77-byte-rpob_signature_message>,
+                                        <16-byte-rpob_hash_tag>
+                                        )
+
+rpob_hash_tag:                16-bytes, utf-8-hash_tag ("rpob/BTC/MAINNET")
+                                    or, utf-8-hash_tag ("rpob/BTC/TESTNET")
+
+rpob_signature_message:       77-bytes, < 1-byte-version_flags>
+                                        <32-byte-input_txid_hash>
+                                        < 4-byte-input_index>
+                                        < 8-byte-burn_output_value>
+                                        <32-byte-rpob_revocation_puzzle>
+</pre>
+
+*The <code>rpob_schnorr_signature</code> uses the [https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki BIP-340] algorithm.
+
+==Deployment==
+
+While the ''IsStandard'' rules are quite restricting, it is quite possible to submit transactions directly to miners, or co-operate with miners who would enjoy to have some addition fee-revenue. So the initial process of testing on the main-network should be possible.
+
+If this standard gains significant attention, we are happy to write a supplementary BIP to define a new service bit to allow nodes to signal that this new type of standard transaction is accepted.
+
+==Rationale==
+===Why require RPoB transactions to use a new transaction version?===
+
+This allows for future indexing and enforcement strategies at the consensus layer.
+
+===Why use a hash puzzle for the revocation?===
+
+The hash-puzzle allows for easy indexing of the revocation, in the format:
+
+<code><32-byte-revocation-puzzle> (if revoked: <32-byte-revocation-puzzle-preimage>)</code>
+
+*Since the digest algorithm is double sha256, even for a very large number of revocation, this index would be very cheap to verify.
+
+*We envision nodes keeping track of all the revocation-puzzles listed in the blockchain, as 32-byte each, a million RPoB is only 32mb. This can be further optimised.
+
+Additionally, we do not want to confuse a public key, (that can sign), and a revocation-puzzle (that may only revoke).
+
+===Why must the revocation output be of zero value?===
+
+Revocation can be spent by anyone once the revocation pre-image has been published.
```

Comment:
Why does it need to be spent at all? It's a one-way signal, right? There's no double-spending to concern with...

Reply:
We think that this is a very good point and expanded on the topic in the "Other Requirements" section: https://github.com/veleslavs/bips/blob/bip-rpob-tx-template/bip-rpob-tx-template.mediawiki#other-requirements

In summary, including the revocation in the block chain makes them uncensorable, in contrast to a gossip network.

---

> *Revocations are both cheaply indexed and uncensorable: ''Receiving the pre-image to the RPoB revocation signals it's revocation. This is easily indexed and propagated in a gossip network, however gossip networks are relatively  censorable. Using the same pre-image, anyone is able to spend the revocation output, rendering this revocation uncensorable.''